poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -894,80 +888,153 @@ public PowerIQAPIClient createClient(FacilitySoftwareConfig poweriq) {
       return realTimeDatas;
    }
 
-   public List<ValueUnit> getValueUnits(Pdu pdu,
+   public List<ValueUnit> getValueUnits(String pduIDFromPowerIQ, PowerIQAPIClient client,
          HashMap<AdvanceSettingType, String> advanceSettingMap) {
       List<ValueUnit> values = new ArrayList<ValueUnit>();
-      if (pdu == null || pdu.getReading() == null || pdu.getReading().getInletReadings() == null) {
+      Long pduId = Long.parseLong(pduIDFromPowerIQ);
+      List<Outlet> outlets = client.getOutlets(pduId);
+      List<Inlet> inlets = client.getInlets(pduId);
+      if (outlets.isEmpty() && inlets.isEmpty()) {
          return values;
       }
-      double voltage = 0.0;
-      double current = 0.0;
-      double power = 0.0;
-      String time = "";
       String dateFormat = advanceSettingMap.get(AdvanceSettingType.DateFormat);
       String timezone = advanceSettingMap.get(AdvanceSettingType.TimeZone);
       String PDU_AMPS_UNIT = advanceSettingMap.get(AdvanceSettingType.PDU_AMPS_UNIT);
       String PDU_POWER_UNIT = advanceSettingMap.get(AdvanceSettingType.PDU_POWER_UNIT);
       String PDU_VOLT_UNIT = advanceSettingMap.get(AdvanceSettingType.PDU_VOLT_UNIT);
-      String TEMPERATURE_UNIT = advanceSettingMap.get(AdvanceSettingType.TEMPERATURE_UNIT);
-      String HUMIDITY_UNIT = advanceSettingMap.get(AdvanceSettingType.HUMIDITY_UNIT);
```

Comment:
Why we remove the humidity and temperature?

Reply:
It's unused in the method, we have  an another job to sync realtime data for sensor